bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/handler/WemoHandler.java

```diff
@@ -73,16 +73,10 @@
 
     private final Map<String, String> stateMap = Collections.synchronizedMap(new HashMap<>());
 
-    // protected static final int SUBSCRIPTION_DURATION = WemoBindingConstants.SUBSCRIPTION_DURATION;
-
-    private UpnpIOService service;
+    protected UpnpIOService service;
+    private WemoHttpCall wemoCall;
 
-    /**
-     * The default refresh interval in Seconds.
-     */
-    private final int DEFAULT_REFRESH_INTERVAL = 120;
-
-    private ScheduledFuture<?> refreshJob;
+    private @Nullable ScheduledFuture<?> refreshJob;
 
     private final Runnable refreshRunnable = new Runnable() {
```

Comment:
You can reduce the code by using a lambda instead of a Runnable:

```java
() -> {
    /* code that is in the run method of the Runnable */
};

Reply:
Would prefer to leave it as si for now, cause never used a lambda here.